tuner/tuner/dispatch_constraints.py

```diff
@@ -25,10 +25,21 @@ def get_mfma_intrinsic_constraints(
 ) -> z3.BoolRef:
     compatible_intrinsics = get_compatible_mfma_intrinsics(problem_size, mma_intrinsics)
     assert len(compatible_intrinsics) > 0, "No compatible intrinsics found"
+
+    mma_attrs = [iree_gpu.MMAAttr.get(mfma) for mfma in compatible_intrinsics]
+    mnk_shapes = [mma_attr.mnk_shape for mma_attr in mma_attrs]
+
     return z3.Or(
         *(
-            z3.And(intrinsic_m == mfma.m, intrinsic_n == mfma.n, intrinsic_k == mfma.k)
-            for mfma in compatible_intrinsics
+            z3.And(
+                intrinsic_m == mnk[0],
+                intrinsic_n == mnk[1],
+                intrinsic_k == mnk[2],
+            )
+            for mma_attr in (
+                iree_gpu.MMAAttr.get(mfma) for mfma in compatible_intrinsics
+            )
+            for mnk in mnk_shapes
```

Comment:
I'm still confused by this loop structure. Why do we need two `for`s?

Reply:
oh, the code should be updated, sorry.